docs/proposals/20230123-azure-service-operator.md

```diff
@@ -0,0 +1,232 @@
+---
+title: Managing Azure Resources with Azure Service Operator
+authors:
+  - "@nojnhuh"
+reviewers:
+  - "@CecileRobertMichon"
+  - "@jackfrancis"
+  - "@matthchr"
+  - "@devigned"
+creation-date: 2023-01-23
+last-updated: 2023-02-21
+status: provisional
+see-also:
+  - https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/416
+  - "/docs/proposals/20210716-async-azure-resource-creation-deletion.md"
+---
+
+# Managing Azure Resources with Azure Service Operator
+
+## Table of Contents
+
+- [Managing Azure Resources with Azure Service Operator](#managing-azure-resources-with-azure-service-operator)
+  - [Table of Contents](#table-of-contents)
+  - [Glossary](#glossary)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals/Future Work](#non-goalsfuture-work)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+      - [Story 1 - Users of new Azure features](#story-1---users-of-new-azure-features)
+      - [Story 2 - Developer experience](#story-2---developer-experience)
+    - [Implementation Details/Notes/Constraints](#implementation-detailsnotesconstraints)
+        - [Reconciliation](#reconciliation)
+        - [API Changes](#api-changes)
+        - [Installation](#installation)
+        - [Caveats](#caveats)
+    - [Security Model](#security-model)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Alternatives](#alternatives)
+      - [Adopt new SDK directly](#adopt-new-sdk-directly)
+      - [Crossplane](#crossplane)
+  - [Upgrade Strategy](#upgrade-strategy)
+  - [Additional Details](#additional-details)
+    - [Test Plan](#test-plan)
+    - [Graduation Criteria](#graduation-criteria)
+  - [Implementation History](#implementation-history)
+
+## Glossary
+
+- **Azure Service Operator (ASO)** - ([link](https://azure.github.io/azure-service-operator/)) a collection of Kubernetes CRDs and controllers which extend the Kubernetes API to manage Azure resources like virtual machines and resource groups.
+- **Bring-your-own (BYO) resource** - an Azure resource backing a Cluster API cluster but managed by the user. CAPI and CAPZ should not modify or delete this resource.
+
+## Summary
+
+This proposal describes how CAPZ could utilize Azure Service Operator (ASO) to create, read, update, and delete Azure resources with a Kubernetes-native interface.
```

Comment:
Kubernetes-native ineterface? Does this mean -- declaring Azure resources in a custom resource that is reconciled by ASO. 
If yes, it would be good to add something like that which imparts the idea of k8s custom resources.

Reply:
Yep, that's right. I'll clarify here.